test/OpenSpaceToolkit/Astrodynamics/Flight/SatelliteSystem.test.cpp

```diff
@@ -0,0 +1,533 @@
+
+/// @project        Open Space Toolkit ▸ Astrodynamics
+/// @file           OpenSpaceToolkit/Astrodynamics/Flight/SatelliteSystem.test.cpp
+/// @author         Antoine Paletta <antoine.paletta@loftorbital.com>
+/// @license        Apache License 2.0
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#include <OpenSpaceToolkit/Astrodynamics/Flight/SatelliteSystem.hpp>
+
+#include <OpenSpaceToolkit/Physics/Environment/Objects/CelestialBodies/Earth.hpp>
+#include <OpenSpaceToolkit/Physics/Environment.hpp>
+#include <OpenSpaceToolkit/Physics/Time/DateTime.hpp>
+#include <OpenSpaceToolkit/Physics/Time/Interval.hpp>
+#include <OpenSpaceToolkit/Physics/Time/Duration.hpp>
+#include <OpenSpaceToolkit/Physics/Time/Instant.hpp>
+#include <OpenSpaceToolkit/Physics/Time/Scale.hpp>
+#include <OpenSpaceToolkit/Physics/Units/Derived/Angle.hpp>
+#include <OpenSpaceToolkit/Physics/Units/Derived.hpp>
+#include <OpenSpaceToolkit/Physics/Units/Length.hpp>
+
+#include <OpenSpaceToolkit/Mathematics/Objects/Vector.hpp>
+#include <OpenSpaceToolkit/Mathematics/Geometry/3D/Objects/Cuboid.hpp>
+
+#include <OpenSpaceToolkit/Core/Containers/Table.hpp>
+#include <OpenSpaceToolkit/Core/Containers/Array.hpp>
+#include <OpenSpaceToolkit/Core/Types/Real.hpp>
+#include <OpenSpaceToolkit/Core/Types/Shared.hpp>
+#include <OpenSpaceToolkit/Core/Types/String.hpp>
+#include <OpenSpaceToolkit/Core/Types/Integer.hpp>
+
+#include <Global.test.hpp>
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+TEST (OpenSpaceToolkit_Astrodynamics_Flight_SatelliteSystem, Constructor)
+{
+
+    using ostk::core::types::Real ;
+    using ostk::core::types::String ;
+    using ostk::core::types::Integer ;
+
+    using ostk::math::obj::Matrix3d ;
+    using ostk::math::geom::d3::objects::Cuboid ;
+    using ostk::math::obj::Vector3d ;
+    using ostk::math::geom::d3::objects::Point ;
+
+    using ostk::physics::units::Mass ;
+
+    using ostk::astro::flight::SatelliteSystem ;
+
+    {
+
+        // Define satellite mass
+        const Mass satelliteMass(90.0, Mass::Unit::Kilogram) ;
+
+        // Define satellite intertia tensor
+        const Matrix3d satelliteInertiaTensor ;
+
+        // Define satellite geometry
+        const Point center = { 0.0, 0.0, 0.0 } ;
+        const std::array<Vector3d, 3> axes = { Vector3d { 1.0, 0.0, 0.0 }, Vector3d { 0.0, 1.0, 0.0 }, Vector3d { 0.0, 0.0, 1.0 } } ;
+        const std::array<Real, 3> extent = { 1.0, 2.0, 3.0 } ;
+
+        const Cuboid satelliteGeometry = {center, axes, extent} ;
+
+        // Define satellite coefficient of drag
+        const Real dragCoefficient = 1.2 ;
+
```

Comment:
@Derollez @robinpdm Is it acceptable that right now you can't only initialize the system with a drag coefficient due to the expected order of the default header initializer? Or is this unnacceptable behavior for a library that others are going to use?